### ISSUE TITLE
Improve documentation

### ISSUE THREAD

**Issue by manuelma** [Spine-project/Spine-Database-API#276]
We should improve the documentation towards the 0.8 release.
We should try to split public and private API, explain why things are needed, add some introduction to the top-level package summarizing the functionality so the user knows where to go.

- [ ] Document the main package
- [x] Document the spineio package
- [x] Document the import_mapping package
- [x] Document the export_mapping package
- [x] Document the filters package
- [x] Make sure only the public API is documented
- [x] Add more examples
- [x] Finalize the tutorial
- [x] Add more tutorials if needed
- [ ] Try to collect feedback on the tutorials, improve

**Comment by soininen** [comment]
The server stuff needs to be documented as well. Not the API, but how it works within Toolbox Tools. When you get the server URL as a command line argument in a Tool script, it is bit different from "raw" database URL, e.g. the database has filters applied already and you cannot add more to the URL (or maybe this is unintentional bug?). You can apply filters using `DatabaseMapping` methods, though.

**Comment by manuelma** [comment]
You're right. There is a server request called apply_filters for that, but I haven't documented it yet. Have you checked the current documentation? what do you think? https://spine-database-api.readthedocs.io/en/0.8-dev/autoapi/spinedb_api/spine_db_server/index.html

**Comment by jkiviluo** [comment]
This is mostly done (although the names refer to some old stuff). What's still missing is something more generic in the beginning (https://github.com/spine-tools/Spine-Toolbox/issues/1991).

And maybe @aporrasc could comment if there is something difficult about the documentation - I think he has been using it in recent weeks.
